src/components/WalletPopup/WalletAddressSelector.js

```diff
@@ -79,12 +81,13 @@ const useGetWallets = () => {
 						},
 					});
 				}
-
-				dispatch(
-					updateWalletReducer({
-						availableWallets: [...availableWallets, ...nextWalletsWithBalances],
-					})
-				);
+				if (mounted) {
+					dispatch(
+						updateWalletReducer({
+							availableWallets: [...availableWallets, ...nextWalletsWithBalances],
+						})
+					);
+				}
 			} catch (e) {
```

Comment:
The `catch` needs `setIsLoading(false);` too right? If `await snxJSConnector.signer.getNextAddresses()` fails `loading` will still be `true` I believe.

Reply:
Nice catch, yeah I believe you are right. I don't think I have ever seen a bug related to this (because the error takes over the loading..), but regardless, it should be `false` here too.